### ISSUE TITLE
ttnn.fill_implicit_tile_padding corrupts BLOCK_SHARDED tensors (and does not zero their padding)

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-metal#53110]
**`ttnn.fill_implicit_tile_padding` is correct on INTERLEAVED tensors and wrong on BLOCK_SHARDED ones**: it fails to zero the implicit tile padding *and* overwrites the logical region.

### Observed

Same tensor, same call, only the memory config differs. `N=1, C=256`, logical `H*W=100` over a padded span of 128 rows, padding pre-filled with `7.0`. Block-sharded is grid 2x2, `COL_MAJOR`, shard `(64, 128)`.

```
INTERLEAVED     logical region max err = 0.0000   padding non-zero =   0.0%  absmax = 0.0000
BLOCK_SHARDED   logical region max err = 0.9961   padding non-zero =  50.0%  absmax = 7.0000
```

### Error (no exception — silently wrong output)

On BLOCK_SHARDED, two things go wrong at once:

- the logical region is **overwritten** — max abs error `0.9961` against the input data that should have been left untouched;
- the padding it was asked to clear is still **50% non-zero**, absmax `7.0`, i.e. the original dirty fill survives.

Likely cause: `fill_implicit_tile_padding` reshapes to rank 3 and dispatches `prim::fill_pad`, which addresses the padding from the **logical** shape. On a block-sharded tensor the padding tail lives on the last M-core, so those offsets land on the wrong cores — writing over real data while missing the rows it meant to clear.

### Expected

The implicit tile-padding rows are zeroed and the logical region is bit-unchanged, for both memory configs — i.e. BLOCK_SHARDED should read `logical region max err = 0.0000, padding non-zero = 0.0%`, as INTERLEAVED already does.

### Steps (exact commands)

```python
# ttnn.fill_implicit_tile_padding on a BLOCK_SHARDED tensor: should zero the tile
# padding and leave the logical rows alone. Interleaved does; block-sharded does neither.
import torch, ttnn

def check(device, sharded):
    torch.manual_seed(0)
    N, C, HW, padded = 1, 256, 100, 128
    logical = torch.rand((N, 1, HW, C), dtype=torch.bfloat16)
    buf = torch.zeros((N, 1, padded, C), dtype=torch.bfloat16)
    buf[:, :, :HW, :] = logical
    buf[:, :, HW:, :] = 7.0

    tt = ttnn.from_torch(buf, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                         device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG)
    tt = ttnn.reshape(tt, ttnn.Shape([N, 1, HW, C]), ttnn.Shape([N, 1, padded, C]))

    if sharded:
        gx, gy = 2, 2
        grid = ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(gx-1, gy-1))})
        spec = ttnn.ShardSpec(grid, (padded // gx, C // gy), ttnn.ShardOrientation.COL_MAJOR)
        cfg = ttnn.MemoryConfig(ttnn.types.TensorMemoryLayout.BLOCK_SHARDED,
                                ttnn.types.BufferType.L1, spec)
        tt = ttnn.to_memory_config(tt, cfg)

    tt = ttnn.fill_implicit_tile_padding(tt, 0.0)

    if sharded:
        tt = ttnn.to_memory_config(tt, ttnn.DRAM_MEMORY_CONFIG)
    out = ttnn.to_torch(ttnn.from_device(ttnn.reshape(tt, ttnn.Shape([N, 1, padded, C])))).float()

    print(f"{'BLOCK_SHARDED' if sharded else 'INTERLEAVED':14s}  "
          f"logical region max err = {(out[:, :, :HW, :] - logical.float()).abs().max().item():.4f}   "
          f"padding non-zero = {(out[:, :, HW:, :].abs() > 0).float().mean().item()*100:5.1f}%  "
          f"absmax = {out[:, :, HW:, :].abs().max().item():.4f}")

device = ttnn.open_device(device_id=0, l1_small_size=8192)
try:
    check(device, sharded=False)
    check(device, sharded=True)
finally:
    ttnn.close_device(device)
```

### Impact

Any caller passing a block-sharded, non-tile-aligned tensor gets silently wrong results, with no exception raised. Measured downstream: applying this op inside `group_norm` took `test_group_norm_block_sharded_non_tile_aligned` from 6 passed to 3 failed, which is what ruled the zero-fill approach out of #52685 and forced the in-kernel masking used in #52924.

### Software Versions

- Ubuntu 24.04
- tt-metal @ `556a284a809` (current main)
- Reproduced on Wormhole B0; originally observed on Blackhole as well
- Split out of #52685 (Defect 2); raised in review of #52924
